tests/longhaul/base_block_test.go

```diff
@@ -0,0 +1,278 @@
+
+	//create mysql pod
+	if _, err := kh.GetPVCStatus(defaultNamespace, pvcName); err != nil {
+		logger.Infof("Create PVC")
+
+		mySqlPodOperation(kh, storageClassName, appName, appLabel, pvcName, "create")
+
+		//wait till mysql pod is up
+		require.True(t(), kh.IsPodInExpectedState(appLabel, "", "Running"))
+		require.True(t(), kh.WaitUntilPVCIsBound(defaultNamespace, pvcName))
+	}
+	dbIP, err := kh.GetPodHostID(appLabel, "")
+	require.Nil(t(), err)
+	dbPort, err := kh.GetServiceNodePort(appName, "default")
+	require.Nil(t(), err)
+	//create database connection
+	db := utils.CreateNewMySQLHelper("mysql", "mysql", dbIP+":"+dbPort, "sample")
+
+	require.True(t(), db.PingSuccess())
+
+	if exist := db.TableExists(); !exist {
+		db.CreateTable()
+	}
+
+	return db
+
+}
+
+func storageClassOperation(k8sh *utils.K8sHelper, storageClassName string, poolName string, namespace string, action string) (string, error) {
+	config := map[string]string{
+		"storageClassName": storageClassName,
+		"poolName":         poolName,
+		"namespace":        namespace,
+	}
+
+	result, err := k8sh.ResourceOperationFromTemplate(action, GetStorageClassDef(), config)
+
+	return result, err
+
+}
+func mySqlPodOperation(k8sh *utils.K8sHelper, storageClassName string, appName string, appLabel string, pvcName string, action string) (string, error) {
+	config := map[string]string{
+		"appName":          appName,
+		"appLabel":         appLabel,
+		"storageClassName": storageClassName,
+		"pvcName":          pvcName,
+	}
+
+	result, err := k8sh.ResourceOperationFromTemplate(action, GetMySqlPodDef(), config)
+
+	return result, err
+
+}
+
+func createPVCOperation(k8sh *utils.K8sHelper, storageClassName string, pvcName string) (string, error) {
+	config := map[string]string{
+		"storageClassName": storageClassName,
+		"pvcName":          pvcName,
+	}
+
+	result, err := k8sh.ResourceOperationFromTemplate("create", getPvcDefinition(), config)
+
+	return result, err
+
+}
+
+func mountUnmountPVCOnPod(k8sh *utils.K8sHelper, podName string, pvcName string, readonly string, action string) (string, error) {
+	config := map[string]string{
+		"podName":  podName,
+		"pvcName":  pvcName,
+		"readOnly": readonly,
+	}
+
+	result, err := k8sh.ResourceOperationFromTemplate(action, getBlockPodDefintion(), config)
+
+	return result, err
+}
+
+func performBlockOperations(installer *installer.InstallHelper, db *utils.MySQLHelper) {
+	var wg sync.WaitGroup
+	for i := 1; i <= installer.Env.LoadConcurrentRuns; i++ {
+		wg.Add(1)
+		go dbOperation(db, &wg)
+	}
+	wg.Wait()
+}
+
+func dbOperation(db *utils.MySQLHelper, wg *sync.WaitGroup) {
+	defer wg.Done()
+	//InsertRandomData
+	db.InsertRandomData()
+	db.InsertRandomData()
+	db.InsertRandomData()
+	db.SelectRandomData(5)
+	db.InsertRandomData()
+	db.InsertRandomData()
+	db.InsertRandomData()
+	db.SelectRandomData(10)
+
+	//delete Data
+	db.DeleteRandomRow()
+	db.SelectRandomData(20)
+
+}
+
+func GetStorageClassDef() string {
```

Comment:
They are a bit different.

Reply:
i actually see three copies in the test tree of the yaml that looks the same, in base_block_test.go, block_createBlock_k8s_test.go, and block_test.go. They all create a pool with replication:1 and the storage class. Looks like the only difference is parameter substitution. Can't these be factored into a helper method to avoid the duplication?

Reply:
i am going to handle this saperatly in https://github.com/rook/rook/issues/1108. because some tests will need refactoring which is out of scope for this PR.